File: Easy_C++/Reverse_Words/Reverse_Words.py
import sys
test_cases = open(sys.argv[1], 'r')
for test in test_cases:
    # ignore test if it is an empty line
    # 'test' represents the test case, do something with it
    if not test in ['\n', '\r\n']:
        test = test.rstrip('\r\n')
        fields = test.split()
        for x in range(len(fields) - 1, -1, -1):
            if x == 0:
                sys.stdout.write(fields[x] + ' ')
            else:
                #print fields[x]
                sys.stdout.write(fields[x] + ' ')
        sys.stdout.write('\n')
test_cases.close()

File: Easy_C++/Reverse_Words/Reverse_Words.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <sstream>
#include <vector>
#include <algorithm>

using namespace std;

vector<string> &split(const string &s, char delim, vector<string> &elms)
{
    stringstream ss(s);
    string item;
    while(getline(ss, item, delim))
    {
        elms.push_back(item);
    }
    return elms;
}

vector<string> split(const string &s, char delim)
{
    vector<string> elms;
    split(s, delim, elms);
    return elms;
}

int main(int argc, char *argv[])
{
    ifstream file;
    file.open(argv[1]);
    string lineBuffer;
    while (!file.eof()) 
    {
        getline(file, lineBuffer);
        if (lineBuffer.length() == 0)
        {
            continue; //ignore all empty lines
        }
        else 
        {
            lineBuffer.erase( std::remove(lineBuffer.begin(), lineBuffer.end(), '\r'), lineBuffer.end());
            vector<string> fields = split(lineBuffer, ' ');
            for(vector<string>::iterator i = fields.end() - 1; i >= fields.begin(); --i)
            {
                cout << *i << " ";
            }
        }
        cout << endl;
    }
}





